ext/http/00_serve.js

```diff
@@ -477,8 +478,27 @@ function mapToCallback(context, callback, onError) {
       return;
     }
 
-    const status = inner.status;
-    const headers = inner.headerList;
+    let status;
+    let headers;
+    let body;
+
+    const inner = toInnerResponse(response);
+    if (inner) {
+      status = inner.status;
+      headers = inner.headerList;
+      body = inner.body;
+    } else {
+      // response is a ServeHandlerResponse
+      status = response.status ?? 200;
```

Comment:
I'm not doing any input validation, { status: 'foo' } will crash the server, similar to what happens now with `Deno.serve(() => null)`. Using `ServeHandlerResponse` is supposed to be a lower level/fast API so I think we shouldn't validate the input.

Reply:
I think we'll need to do some very basic validation in this codepath, but it should be pretty minimal. The challenge here is that there's no easy way for the developer to get feedback on failures.

Reply:
Added status validation and tests for it.

```
Terminating Deno.serve loop due to unexpected error RangeError: The status provided (50) is not equal to 101 and outside the range [200, 599].
    at ext:deno_http/00_serve.js:502:17
    at eventLoopTick (ext:core/01_core.js:183:11)
Terminating Deno.serve loop due to unexpected error TypeError: Invalid status (foo)
    at ext:deno_http/00_serve.js:498:17
    at eventLoopTick (ext:core/01_core.js:183:11)
Terminating Deno.serve loop due to unexpected error TypeError: Invalid status ([object Object])
    at ext:deno_http/00_serve.js:498:17
    at eventLoopTick (ext:core/01_core.js:183:11)
Terminating Deno.serve loop due to unexpected error RangeError: The status provided (600) is not equal to 101 and outside the range [200, 599].
    at ext:deno_http/00_serve.js:502:17
    at eventLoopTick (ext:core/01_core.js:183:11)
Terminating Deno.serve loop due to unexpected error RangeError: The status provided (100) is not equal to 101 and outside the range [200, 599].
    at ext:deno_http/00_serve.js:502:17
    at eventLoopTick (ext:core/01_core.js:183:11)
Terminating Deno.serve loop due to unexpected error TypeError: Invalid status (200)
    at ext:deno_http/00_serve.js:498:17
    at eventLoopTick (ext:core/01_core.js:183:11)
 ```